From 48d9bde876ab5d51930f8f9e462c0d25e73475ed Mon Sep 17 00:00:00 2001
From: Marcelo Leitner <mleitner@redhat.com>
Date: Tue, 21 Oct 2014 16:17:12 -0400
Subject: [net] netfilter: nf_tables: set names cannot be larger than 15 bytes

Message-id: <20141021161712.22888.86984.stgit@localhost.localdomain>
Patchwork-id: 97931
O-Subject: [RHEL7 net PATCH 034/124] netfilter: nf_tables: set names cannot be larger than 15 bytes
Bugzilla: 1148041 1155088
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

Backport of upstream commit: a9bdd8365684810e3de804f8c51e52c26a5eccbb

 commit a9bdd8365684810e3de804f8c51e52c26a5eccbb
 Author: Pablo Neira Ayuso <pablo@netfilter.org>
 Date:   Mon Mar 24 15:10:37 2014 +0100

    netfilter: nf_tables: set names cannot be larger than 15 bytes

    Currently, nf_tables trims off the set name if it exceeeds 15
    bytes, so explicitly reject set names that are too large.

    Reported-by: Giuseppe Longo <giuseppelng@gmail.com>
    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>

Signed-off-by: Marcelo Ricardo Leitner <mleitner@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/netfilter/nf_tables_api.c b/net/netfilter/nf_tables_api.c
index 60feca9..3ff4a80 100644
--- a/net/netfilter/nf_tables_api.c
+++ b/net/netfilter/nf_tables_api.c
@@ -1985,7 +1985,8 @@ nft_select_set_ops(const struct nlattr * const nla[],
 
 static const struct nla_policy nft_set_policy[NFTA_SET_MAX + 1] = {
  [NFTA_SET_TABLE]  = { .type = NLA_STRING },
- [NFTA_SET_NAME]   = { .type = NLA_STRING },
+ [NFTA_SET_NAME]   = { .type = NLA_STRING,
+         .len = IFNAMSIZ - 1 },
  [NFTA_SET_FLAGS]  = { .type = NLA_U32 },
  [NFTA_SET_KEY_TYPE]  = { .type = NLA_U32 },
  [NFTA_SET_KEY_LEN]  = { .type = NLA_U32 },
-- 
1.7.1